stdlib/tests/mast_forest_merge.rs

```diff
@@ -0,0 +1,18 @@
+use processor::MastForest;
+
+/// Tests that the stdlib merged with itself produces a forest that has the same procedure
+/// roots.
+///
+/// This test is added here since we do not have the StdLib in miden-core where merging is
+/// implemented and the StdLib serves as a convenient example of a large MastForest.
+#[test]
+fn mast_forest_merge_stdlib() {
+    let std_forest = miden_stdlib::StdLibrary::default().mast_forest().as_ref().clone();
+
+    let (merged, _) = MastForest::merge([&std_forest, &std_forest]).unwrap();
+
+    let merged_digests = merged.procedure_digests().collect::<Vec<_>>();
+    for digest in std_forest.procedure_digests() {
+        assert!(merged_digests.contains(&digest));
+    }
+}
```

Comment:
I tried this first, but yes, they are not exactly equivalent because the DFS iteration seems to produce a somewhat different ordering. Since there are multiple valid topological orderings for a DAG, this is not a problem. So I just assert that all procedures are still there, which should be good enough.